app/models/concerns/edition/scopes/filterable_by_topical_event.rb

```diff
@@ -0,0 +1,11 @@
+module Edition::Scopes::FilterableByTopicalEvent
```

Comment:
I wonder if there's an opportunity to declare a module like `Edition::HasTopicalEvents` or `Edition::FeaturesTopicalEvents` which includes both this scope and the `has_many` relationship definition? Then we'd have a nicely encapsulated domain concern.

Not sure whether that can also be applied to other of these scope modules that we've separated out, e.g. maybe one for `Edition::ChecksBrokenLinks` etc.?

In an ideal world I'd love to decouple such concerns from the `Edition` model but I expect that could only work if we used polymorphic relationships for topical event featurings etc. and may introduce other complexities (I think that's where the document relationship broke down last time you tried iirc)

Reply:
I did try adding `with_topical_event` and `in_world_location` to the `Edition::TopicalEvent` and `Edition::WorldLocation` concerns, but as there is filtering on the main `Edition` model (i.e `Edition.with_topical_event`) those filters would only be available on models that have that module included, hence why I moved it to a Scope module.

There's definitely scope for some of the other concerns though, like broken links. I'll give that a go 👍

Reply:
I think "those filters would only be available on models that have that module included" is ok, isn't it? We could just include it on the `Edition` model rather than the individual edition type models?

I know we'd like to get away from that eventually, but I think the right way to go is to acknowledge where, because of the way we consume the behaviours in the application (e.g. we don't check that we're in a context where we can expect to be able to check if the edition is associated with topical events), we can't decouple edition types from each other.  We can eventually work towards enabling decoupling, but in the mean time having separate scope concerns to work around that feels a little bit artificial to me. What do you think?